File: GenerateLinqDB/GenerateVB/Flow/OracleGenerateFlow.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Text
Imports GenerateVB.Utilities

Namespace Flow
    Public Class OracleGenerateFlow
        Inherits Flow.BaseGenerateFlow

        Private _dal As OracleGenerateDAL

        Private ReadOnly Property DALObj() As OracleGenerateDAL
            Get
                If _dal Is Nothing Then
                    _dal = New OracleGenerateDAL()
                End If
                Return _dal
            End Get

        End Property

        Private Sub SetData(ByVal Data As GenerateVB.Data.GenerateData)
            DALObj.DataSource = Data.DataSource
            DALObj.DataBaseName = Data.DataBaseName
            DALObj.UserID = Data.UserID
            DALObj.Password = Data.Password
            DALObj.TableName = Data.TableName
            DALObj.DatabaseType = Data.DatabaseType
            _IsView = DALObj.IsView()
            _databaseType = DALObj.DatabaseType
            _columnTable = DALObj.GetTableColumn()
            _pkColumnTable = DALObj.GetPKColumn()
            _uniqueColumnTable = DALObj.GetUniqueColumn()
            _tableName = DALObj.TableName

        End Sub

        Private Sub SetConnDesc(ByVal Data As GenerateVB.Data.GenerateData)
            DALObj.DataSource = Data.DataSource
            DALObj.DataBaseName = Data.DataBaseName
            DALObj.UserID = Data.UserID
            DALObj.Password = Data.Password
            DALObj.DatabaseType = Data.DatabaseType
            _databaseType = DALObj.DatabaseType
        End Sub

        Public Function GenerateCodeDAL(ByVal Data As Data.GenerateData) As String
            SetData(Data)
            Return GenerateLinq(Data)
        End Function

        Public Function GenerateCodeData(ByVal Data As Data.GenerateData) As String
            SetData(Data)
            Return GeneratePara(Data)
        End Function


        Public Function GetTableList(ByVal Data As Data.GenerateData) As DataTable
            SetConnDesc(Data)
            Return DALObj.GetTableList()
        End Function

        Public Function GetAllTable(ByVal Data As Data.GenerateData) As DataTable
            SetConnDesc(Data)
            Return DALObj.GetAllTable()
        End Function
    End Class
End Namespace

File: GenerateLinqDB/GenerateC/Flow/OracleGenerateFlow.cs
using System;
using System.Collections.Generic;
using System.Text;
using GenerateC.Utilities;
using GenerateC.Data;
using System.Data;

namespace GenerateC.Flow
{
    public class OracleGenerateFlow : BaseGenerateFlow
    {
        OracleGenerateDAL _dal;

        private OracleGenerateDAL DALObj
        {
            get { if (_dal == null) { _dal = new OracleGenerateDAL(); } return _dal; }
        }

        private void SetData(GenerateData data)
        {
            DALObj.DataSource = data.DataSource;
            DALObj.DatabaseName = data.DatabaseName;
            DALObj.UserID = data.UserID;
            DALObj.Password = data.Password;
            DALObj.TableName = data.TableName;
            _tableName = data.TableName;
            _databaseType = DALObj.DatabaseType;
            _isView = DALObj.IsView();
            _columnTable = DALObj.GetTableColumn();
            
            if (!_isView)
            {
                _uniqueColumnTable = DALObj.GetUniqueColumn();
                _pkColumnTable = DALObj.GetPKColumn();
            }
        }
        private void SetConnDesc(GenerateC.Data.GenerateData data)
        {
            DALObj.DataSource = data.DataSource;
            DALObj.DatabaseName = data.DatabaseName;
            DALObj.UserID = data.UserID;
            DALObj.Password = data.Password;
            DALObj.DatabaseType = data.DatabaseType;
            _databaseType = DALObj.DatabaseType;
        }

        public string GenerateLinq(GenerateData data)
        {
            SetData(data);
            return GenerateLinqCode(data);
        }

        public string GeneratePara(GenerateData data)
        {
            SetData(data);
            return GenerateParaCode(data);
        }

        public DataTable GetTableList(Data.GenerateData Data)
        {
            SetConnDesc(Data);
            return DALObj.GetTableList();
        }
    }
}
